packages/capabilities/src/types.ts

```diff
@@ -438,18 +438,34 @@ export type StoreRemove = InferInvokedCapability<typeof StoreCaps.remove>
 export type StoreList = InferInvokedCapability<typeof StoreCaps.list>
 
 export type StoreAddSuccess = StoreAddSuccessDone | StoreAddSuccessUpload
-export interface StoreAddSuccessDone {
-  status: 'done'
+
+export type StoreAddSuccessStatusUpload = 'upload'
+export type StoreAddSuccessStatusDone = 'done'
+
+export interface StoreAddSuccessResult {
+  /**
+   * Status of the item to store. A "done" status indicates that it is not
+   * necessary to upload the item. An "upload" status indicates that the item
+   * should be uploaded to the provided URL.
+   */
+  status: StoreAddSuccessStatusUpload | StoreAddSuccessStatusDone
+  /**
+   * Total bytes allocated in the space to accommodate this stored item.
+   * May be zero if the item is _already_ stored in _this_ space.
+   */
+  allocated: number
```

Comment:
We may want to advertise how much we can ensure consistency of this, which I imagine might be 'not very much'.

e.g. let's say cid A is not stored in the space. Two `store/add` invocations get created with different CIDs (e.g. different nonce/expirty) to add cid A to the space. Then those two invocations are submitted to us at the exact same second. iiuc it may be the case that both of those responses come back with `allocated: size(A)` (because we dont have transactions, and in this scenario both invocations calls `carStoreBucket.has(link)` [here](https://github.com/web3-storage/w3up/blob/880f0013049862658e07038016b647c323920f16/packages/upload-api/src/store/add.js#L37) would have returned false.

my main concern
* someone might use this `allocated` response value in some business logic without being aware of the eventual consistency lack-of-transactionality in generating this value. i.e. this could be nonzero even if technically the a different `store/add` invocation added the CID first (in the race condition describes above).
* in that race condition, the same CID may have multiple store/add receipts whose results have `allocated` set to the size of the referent of the CID.

If we want to return this information in the result of the write operation (store/add), we probably need something like [transactions](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/transaction-apis.html) to do it in a way that isn't subject to these race conditions.
But if possible it may be nice to avoid returning this value in the result of the write operation (because it almost always increases the cost of the write operation to add return values that. must also be consistent)

alternatively we ignore the above concerns but maybe it's a good idea to at least put in the jsdoc here that there a nonzero response does not necessarily indicate that the `store/add` is the authoritative addition of the CID to the space.

Reply:
Great point, in hindsight this is maybe not the right solution because _we_ want to use it in _our_ business logic...

Reply:
good point, so maybe we want the insert later on to have a result of being conflict or not. Only one Put should happen in dynamo successfully

Reply:
Ok, on reflection I think we _can_ make `allocated` in the response be authoratitive about the addition of the CID to the space. This is what we can do:

* We don't need `TransactWriteItems` since we're not writing multiple items!
* We can use conditional expressions to ensure writes fail:
    * Change store/add `PutItem` so that it fails if [not a new record](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.ConditionExpressions.html#Expressions.ConditionExpressions.PreventingOverwrites)
    * Change store/remove `DeleteItem` so that it fails if record does not exist